src/providers/personalData/store.tsx

```diff
@@ -0,0 +1,109 @@
+import { createStore } from '@/store'
+import { readFromLocalStorage } from '@/utils/localStorage'
+
+import {
+  DismissedMessage,
+  FollowedChannel,
+  RecentSearch,
+  RecentSearchType,
+  WatchedVideo,
+  WatchedVideoStatus,
+} from './types'
+
+export type PersonalDataStoreState = {
+  watchedVideos: WatchedVideo[]
+  followedChannels: FollowedChannel[]
+  recentSearches: RecentSearch[]
+  dismissedMessages: DismissedMessage[]
+  playerVolume: number
+}
+
+const WHITELIST = [
+  'watchedVideos',
+  'followedChannels',
+  'recentSearches',
+  'dismissedMessages',
+  'playerVolume',
+] as (keyof PersonalDataStoreState)[]
+
+export type PersonalDataStoreActions = {
+  updateWatchedVideos: (__typename: WatchedVideoStatus, id: string, timestamp?: number) => void
+  updateChannelFollowing: (id: string, follow: boolean) => void
+  updateRecentSearches: (id: string, type: RecentSearchType) => void
+  updateDismissedMessages: (id: string, add?: boolean) => void
+  updatePlayerVolume: (volume: number) => void
+}
+
+const watchedVideos = readFromLocalStorage<WatchedVideo[]>('watchedVideos') ?? []
+const followedChannels = readFromLocalStorage<FollowedChannel[]>('followedChannels') ?? []
+const recentSearches = readFromLocalStorage<RecentSearch[]>('recentSearches') ?? []
+const dismissedMessages = readFromLocalStorage<DismissedMessage[]>('dismissedMessages') ?? []
+const playerVolume = readFromLocalStorage<number>('playerVolume') ?? 1
```

Comment:
```suggestion
```

I don't think we should use these. Persist should handle setting up state from localstorage

Reply:
That's the workaround workaround for the migration. Since we're introducing new key in localStorage (`personalData`) - migrate will not fire. That's why I'm populating all personalData-related fields to initial state and then cleaning storage up. Alternative solution would be creating separate store for each field